cmake/TestSetup.cmake

```diff
@@ -11,6 +11,27 @@
 # the public, perform publicly and display publicly, and to permit others to do so.
 #=========================================================================================
 
+
+
+# Search for the python interpreter
+if ( CMAKE_VERSION VERSION_LESS "3.12")
+  # Version number has been intentionally excluded from find_package call, so that latest version 
+  # will be grabbed. Including the version number would prioritise the version provided over more 
+  # up to date versions
+  find_package(PythonInterp REQUIRED)
+  if( ${PYTHON_VERSION_STRING} VERSION_LESS "3.6")
```

Comment:
Is 3.6 the version we want to pin on? I don't object---Python development moves slowly and minor versions don't change very often. Just wanted to make sure there's no users or environments we're missing out on by picking this.

Reply:
I think it is a good choice. Has most modern features.

Reply:
Yeah but are we actually using any of those modern features? We should certainly pin ourselves on version >3, since 2 is deprecated. But I don't want to pick a python version that's missing on, say, an HPC cluster we need. 

That said python 3.6 was first released in 2016, so we're probably fine. I just wanted to make sure. @pgrete, @forrestglines , @jmstone : are there any HPC systems you use that are stuck on an older version of Python? 3.2 might be the safest choice, as that was released in 2011.

Reply:
@pgrete @forrestglines @jmstone

Reply:
I think Python 3.6 is a good choice. It's been some time since I came across an old python 3.x version (all the systems I checked have python >= 3.6 available either by default or through modules)

Reply:
Fine by me then.